src/pages/my-page/postsIdeaMarket/IdeaMarketEdit.tsx

```diff
@@ -1,42 +1,101 @@
-//import QnASection from '../../../components/postdetail/QnASection';
-//import AuthorInfo from '../../../components/postdetail/AuthorInfo';
+import { ProfileHeaderAuthor } from '../../../components/my-page/ProfileHeaderAuthor';
+import PostTitlePay from '../../../components/registeredPage/PostTitlePay';
+import IdeaDescription from '../../../components/registeredPage/IdeaDescription';
+import QnASection from '../../../components/postdetail/QnASection';
+import AuthorInfo from '../../../components/postdetail/AuthorInfo';
+import styles from '../../../pages/idea-market/ideaRegisteredPage.module.scss';
+
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { useQuery } from '@tanstack/react-query';
+import { getIdeaMarketDetail } from '../../../apis/detailPageAPI';
+import { IdeaMarketDetail } from '../../../types/detailPageType';
+import { getUserIdFromToken } from '../../../utils/auth';
 
 export const IdeaMarketEdit = () => {
-  const FORM_DATA = {
-    descriptionTitle: '아이디어 설명',
-    description: '아이디어 설명입니다롱',
-    attachmentTitle: '첨부파일',
-    attachmentFileName: '첨부파일 제목 입니다... .pdf',
+  const { ideaId } = useParams<{ ideaId: string }>();
+  const userId = getUserIdFromToken();
+  const [resolvedIdeaId, setResolvedIdeaId] = useState<number | null>(null);
+  const [isReady, setIsReady] = useState(false);
+  console.log('IdeaMarketEdit.tsx - URL Params ideaId:', ideaId);
+
+  // ideaId가 존재할 때만 resolvedIdeaId를 설정하고 렌더링 진행
+  // 새로고침 시 ideaId가 존재하지 않아 렌더링이 중단되는 것을 방지
+  useEffect(() => {
+    if (ideaId) {
+      console.log('useEffect - ideaId 업데이트됨:', ideaId);
+      setResolvedIdeaId(Number(ideaId));
+      setIsReady(true); // 데이터 준비 완료
+    }
+  }, [ideaId]);
+
+  const {
+    data: post,
+    isLoading,
+    isError,
+  } = useQuery<IdeaMarketDetail, Error>({
+    queryKey: ['ideaMarketDetail', resolvedIdeaId],
+    queryFn: () => getIdeaMarketDetail(resolvedIdeaId!),
+    enabled: isReady,
+    staleTime: 1000 * 60 * 10,
+  });
+
+  console.log('IdeaMarketEdit.tsx - post:', post);
+
+  const postData = {
+    ideaId: Number(ideaId),
+    title: post?.title ?? '',
+    category: post?.category ?? '',
+    ideaMarketType: post?.ideaMarketType ?? '',
+    auth: post?.auth ?? '',
+    price: post?.price ?? 0,
+    viewCount: post?.viewCount ?? 0,
+    saveCount: post?.saveCount ?? 0,
+    thumbnailImageUrl: post?.thumbnailImageUrl ?? '',
+    createdDate: post?.createdDate ?? '',
   };
-  const USER_DATA = {
-    postId: 2,
-    userName: 'SEO YEON',
-    profileImage: null,
+
+  const writerData = {
+    writerId: post?.writer?.writerId ?? 0,
+    name: post?.writer?.name ?? '',
+    profileImageUrl: post?.writer?.profileImageUrl ?? '',
+    role: post?.writer?.role ?? '',
+    specialization: post?.writer?.specialization ?? '',
+    totalIdeas: post?.writer?.totalIdeas ?? 0,
+    totalCollaborations: post?.writer?.totalCollaborations ?? 0,
```

Comment:
필수는 아니지만,, 전 요렇게 하면 더 깔꼼해보이더라구요!

```suggestion
  const writerDataInit = {
    writerId: 0,
    name:'',
    profileImageUrl:  '',
    role:  '',
    specialization:  '',
    totalIdeas:  0,
    totalCollaborations: 0,
}
const { writerId, name, profileImageUrl, role. ..} = writerData ?? writerDataInit
```